test/oqs_test_libctx.c

```diff
@@ -0,0 +1,598 @@
+// SPDX-License-Identifier: Apache-2.0 AND MIT
+
+#include <openssl/core_names.h>
+#include <openssl/evp.h>
+#include <openssl/provider.h>
+#include <openssl/rand.h>
+#include <openssl/types.h>
+#include <string.h>
+
+#include "oqs_prov.h"
+#include "test_common.h"
+
+#define MAX_DUMMY_ENTROPY_BUFFERLEN 0x100000
+
+/** \brief The info about classical elements needed. */
+struct ClassicalInfo {
+    /** \brief Name label. */
+    const char *name;
+
+    /** \brief The public key length, in bytes. */
+    size_t pubkey_len;
+
+    /** \brief The private key length, in bytes. */
+    size_t privkey_len;
+
+    /** \brief The shared secret length, in bytes. */
+    size_t sec_len;
+
+    /** \brief The signature length, in bytes. */
+    size_t sig_len;
+};
+
+static const struct ClassicalInfo info_classical[] = {
+    {"p256", 65, 121, 32, 72},      {"SecP256r1", 65, 121, 32, 72},
+    {"p384", 97, 167, 48, 104},     {"SecP384r1", 97, 167, 48, 104},
+    {"p521", 133, 223, 66, 141},    {"SecP521r1", 133, 223, 66, 141},
+    {"bp256", 65, 122, 32, 72},     {"bp384", 97, 171, 48, 104},
+    {"rsa3072", 398, 1770, 0, 384}, {"pss3072", 398, 1770, 0, 384},
+    {"rsa2048", 270, 1193, 0, 256}, {"pss2048", 270, 1193, 0, 256},
+    {"ed25519", 32, 32, 0, 64},     {"ed448", 57, 57, 0, 114},
+    {"x25519", 32, 32, 32, 0},      {"X25519", 32, 32, 32, 0},
+    {"x448", 56, 56, 56, 0},
+};
+
+static OSSL_LIB_CTX *libctx = NULL;
+static char *modulename = NULL;
+static char *configfile = NULL;
+
+/** \brief Loads OpenSSL's 'TEST-RAND' deterministic pseudorandom generator for
+ * the given library context
+ *
+ * \param libctx Library context.
+ *
+ * \return 1 if the configuration is successfully loaded, else 0. */
+static int oqs_load_det_pseudorandom_generator(OSSL_LIB_CTX *libctx) {
+    OSSL_PARAM params[2], *p = params;
+    unsigned char entropy[MAX_DUMMY_ENTROPY_BUFFERLEN];
+    RAND_bytes(entropy, sizeof(entropy));
+
+    if (!RAND_set_DRBG_type(libctx, "TEST-RAND", NULL, NULL, NULL)) {
+        return 0;
+    }
+
+    *p++ = OSSL_PARAM_construct_octet_string(OSSL_RAND_PARAM_TEST_ENTROPY,
+                                             entropy, sizeof(entropy));
+    *p = OSSL_PARAM_construct_end();
+
+    EVP_RAND_CTX *rctx_public = RAND_get0_public(libctx);
+    if (!rctx_public) {
+        return 0;
+    }
+
+    if (!EVP_RAND_CTX_set_params(rctx_public, params)) {
+        return 0;
+    }
+
+    EVP_RAND_CTX *rctx_private = RAND_get0_private(libctx);
+    if (!rctx_private) {
+        return 0;
+    }
+
+    if (!EVP_RAND_CTX_set_params(rctx_private, params)) {
+        return 0;
+    }
+
+    return 1;
+}
+
+/** \brief Resets the given library context's DRBG instances
+ *
+ * \param libctx Library context.
+ *
+ * \return 1 if the configuration is successfully restarted, else 0. */
+static int oqs_reset_det_pseudorandom_generator(OSSL_LIB_CTX *libctx) {
```

Comment:
Strictly speaking, the parameter is not required here as all operations happen on the file-global variable `libctx` anyway.

So, to ease code readability/understanding, I'd suggest to drop it here and in all other functions -- or to remove the `static`/file-global declaration (and only pass `libctx` as parameter).